packages/server-wallet/src/message-service/test-message-service.ts

```
@@ -42,7 +42,10 @@ export class TestMessageService implements MessageServiceInterface {
     // We always pass a reference to the messageService when calling handleMessage
     // This allows the MessageHandler function to easily call messageHandler.send
     // We just bind that here for convenience.
-    this._handleMessage = async message => handleMessage(message, this);
+    this._handleMessage = async message => {
+      // This prevents triggering messages after the service is destroyed
+      if (!this._destroyed) return handleMessage(message, this);
```

Should we throw an error or something if `this._destroyed`?